packages/wallet/ui/src/components/ConnectionSettingsDialog.jsx

```diff
@@ -1,106 +1,359 @@
+        },
+  );
+
+  const [currentTab, setCurrentTab] = useState(
+    connectionConfig && connectionConfig.type === ConnectionConfigType.Solo
+      ? '1'
+      : '0',
+  );
+
+  const config = currentTab === '0' ? smartWalletConfig : soloWalletConfig;
+
+  const [selectInputId] = useState(uuid());
+
+  const errors = new Set();
+
   try {
-    const url = new URL(connection);
-    isValid = url !== undefined;
+    // eslint-disable-next-line no-unused-vars
+    const url = new URL(config.href);
   } catch (e) {
-    isValid = false;
+    errors.add(Errors.INVALID_URL);
+  }
+
+  if (config.type === ConnectionConfigType.Smart) {
+    if (config.readOnly && !config.publicAddress) {
+      errors.add(Errors.INVALID_ADDRESS);
+    }
+    if (config.readOnly === undefined) {
+      errors.add(Errors.READ_ONLY_UNSPECIFIED);
+    }
+  }
+
+  if (config.type === ConnectionConfigType.Solo && !config.accessToken) {
+    errors.add(Errors.INVALID_ACCESS_TOKEN);
   }
 
   const saveAndClose = () => {
-    if (connection) {
-      setWalletConnection(connection);
+    if (config) {
+      if (config.accessToken) {
+        maybeSave('accessToken', config.accessToken);
+      }
+      setConnectionConfig(config);
       disconnect(true);
-      const isKnown = allWalletConnections.some(c => c === connection);
+      const { href, type } = config;
+      const isKnown = allConnectionConfigs.some(
+        c => c.href === href && c.type === type,
+      );
       if (!isKnown) {
-        setAllWalletConnections(conns => [connection, ...conns]);
+        setAllConnectionConfigs(conns => [{ href, type }, ...conns]);
       }
     }
     onClose();
   };
 
+  const handleTabChange = (_ev, value) => {
+    setCurrentTab(value);
+  };
+
+  const smartWalletConfigForm = (
+    <>
+      <Autocomplete
+        value={smartWalletConfig?.href}
+        id="connection"
+        options={smartConnectionHrefs}
+        sx={{ width: 360, mt: 2 }}
+        onChange={(_, newValue) =>
+          setSmartWalletConfig(swConfig => ({
+            ...swConfig,
+            href: newValue,
+          }))
+        }
+        filterOptions={(options, params) => {
+          const { inputValue } = params;
+          const isExisting = options.some(option => inputValue === option);
+          if (inputValue && !isExisting) {
+            options.unshift(inputValue);
+          }
+          return options;
+        }}
+        renderOption={(props, option) => <li {...props}>{option}</li>}
+        selectOnFocus
+        clearOnBlur
+        handleHomeEndKeys
+        freeSolo
+        renderInput={params => (
+          <TextField {...params} label="Chain Config URL" />
+        )}
+      />
+      <ErrorLabel>
+        {errors.has(Errors.INVALID_URL) ? 'Enter a valid URL' : ''}
+      </ErrorLabel>
+      <FormControl sx={{ mt: 2 }}>
+        <InputLabel id={selectInputId}>Connection Method</InputLabel>
+        <Select
+          value={smartWalletConfig.readOnly}
```

Comment:
this boolean suggests a checkbox control. either make the UI match the boolean or make the config have a string that the menu selects